### PR TITLE
Add tilt support

### PR DESCRIPTION
**Pull request by fiunchinho**
Added tilt support to easily manual test policies.

I've added a sample `cluster.yaml` and `machinepool.yaml` files in the `e2e` folder. These can be used from the Tilt UI and you get back the mutated result.

Also, Tilt deploys kyverno so you can see the kyverno logs from the Tilt UI.

image

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by MarcelMue** [comment]
I like the changes - only now got to reviewing them. Is the failure of test-policies related?

File: .circleci/config.yml [hunk 1]
```diff
@@ -76,17 +76,17 @@ jobs:
           command: |
             sudo apt-get update && sudo apt-get install make
       - run:
-          name: Setup
```
Comment (fiunchinho) [inline .circleci/config.yml:79]: Install clusterctl so we can deploy CAPI components. We need them so we can have the conversion webhooks and have a more realistic test cluster.

File: .circleci/config.yml [hunk 2]
```diff
@@ -76,17 +76,17 @@ jobs:
...
-            make setup
+            curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.4/clusterctl-linux-amd64 -o clusterctl && chmod +x clusterctl  && sudo mv clusterctl /usr/bin
+      - run:
+          name: Create kind cluster and install kyverno
+          command: |
+            make kind-create kind-get-kubeconfig install-kyverno
       - run:
           name: Make chart archives
           command: |
-            dabs.sh --generate-metadata --chart-dir helm/policies-common
```
Comment (fiunchinho) [inline .circleci/config.yml:85]: Moved to `Makefile`.

File: README.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ Tests are implemented with [pytest](https://docs.pytest.org) with plugin [pytest
 Executing the integration tests can be done with this simple set of commands:
 ```bash
 make setup # Creates the kind cluster and installs all dependencies.
-./dabs.sh --generate-metadata -c ./helm/policies-aws # Builds helm chart archive to be tested.
+dabs.sh --generate-metadata -c ./helm/policies-aws # Builds helm chart archive to be tested.
```
Comment (fiunchinho) [inline README.md:29]: needs to be on the `$PATH` now so we do the same locally and during CI.

**Comment by MarcelMue** [comment]
Before I review deeper:
What about the merge conflict?

File: e2e/awscluster.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
```
Comment (fiunchinho) [inline e2e/awscluster.yaml:1]: The files on these folder are meant to be used as a quickly manual test from the Tilt UI.

File: helm/tests/ensure.py [hunk 5]
```diff
@@ -430,7 +433,7 @@ def awscluster(kubernetes_cluster):
     LOGGER.info(f"AWSCluster {cluster_name} applied")
 
     raw = kubernetes_cluster.kubectl(
-        f"get awscluster {cluster_name}", output="yaml")
+        f"get awsclusters.v1alpha3.infrastructure.cluster.x-k8s.io {cluster_name}", output="yaml")
```
Comment (fiunchinho) [inline helm/tests/ensure.py:436]: We are using v1alpha4 as the stored version on etcd, so we need to be explicit about which api version we want so conversion webhooks are called.

**Comment by fiunchinho** [comment]
> Before I review deeper: What about the merge conflict?

should be fine now

**Review by MarcelMue — APPROVED** [review]
HELL YEAH

**Pull request merged**